witchcraft-server/src/witchcraft.rs

```diff
@@ -139,6 +142,14 @@ impl Witchcraft {
                 .map(|e| extend_path(e, self.install_config.context_path(), prefix)),
         )
     }
+
+    /// Adds a future that will be polled until ready when the server shuts down
+    pub fn add_shutdown_hook<F>(&mut self, future: F)
```

Comment:
Do you think this would be better named `on_shutdown`?

The docs could also be a bit more explicit about how the server will wait for a limited amount of time for the hook to complete before shutting down fully.

Reply:
I think that makes sense. The one worry I have is that `on_shutdown` makes it sound like only one can be set? Although I guess it is consistent with other methods like `app` that have similar semantics. Will update